core/src/main/java/org/apache/gravitino/authorization/AuthorizationUtils.java

```diff
@@ -364,4 +376,127 @@ private static void checkCatalogType(
+    try {
+      switch (metadataObject.type()) {
+        case METALAKE:
+          {
+            NameIdentifier[] identifiers =
+                GravitinoEnv.getInstance().catalogDispatcher().listCatalogs(Namespace.of(metalake));
+            Arrays.stream(identifiers)
+                .collect(Collectors.toList())
+                .forEach(
+                    identifier -> {
+                      Catalog catalogObj =
+                          GravitinoEnv.getInstance().catalogDispatcher().loadCatalog(identifier);
+                      if (catalogObj.provider().equals("hive")) {
+                        Schema schema =
+                            GravitinoEnv.getInstance()
+                                .schemaDispatcher()
+                                .loadSchema(
+                                    NameIdentifier.of(
+                                        metalake,
+                                        catalogObj.name(),
+                                        "default" /*Hive default schema*/));
+                        if (schema.properties().containsKey(HiveConstants.LOCATION)) {
+                          String defaultSchemaLocation =
+                              schema.properties().get(HiveConstants.LOCATION);
+                          if (defaultSchemaLocation != null && !defaultSchemaLocation.isEmpty()) {
+                            locations.add(defaultSchemaLocation);
+                          } else {
+                            LOG.warn("Catalog %s location is not found", ident);
+                          }
+                        }
+                      }
+                    });
+          }
+          break;
+        case CATALOG:
+          {
+            Catalog catalogObj = GravitinoEnv.getInstance().catalogDispatcher().loadCatalog(ident);
+            if (catalogObj.provider().equals("hive")) {
+              Schema schema =
+                  GravitinoEnv.getInstance()
+                      .schemaDispatcher()
+                      .loadSchema(
+                          NameIdentifier.of(
+                              metalake, catalogObj.name(), "default" /*Hive default schema*/));
+              if (schema.properties().containsKey(HiveConstants.LOCATION)) {
+                String defaultSchemaLocation = schema.properties().get(HiveConstants.LOCATION);
+                if (defaultSchemaLocation != null && !defaultSchemaLocation.isEmpty()) {
+                  locations.add(defaultSchemaLocation);
+                } else {
+                  LOG.warn("Catalog %s location is not found", ident);
+                }
+              }
+            }
+          }
+          break;
+        case SCHEMA:
+          {
+            Schema schema = GravitinoEnv.getInstance().schemaDispatcher().loadSchema(ident);
+            if (schema.properties().containsKey(HiveConstants.LOCATION)) {
+              String schemaLocation = schema.properties().get(HiveConstants.LOCATION);
```

Comment:
What should we get the location of schema if a schema has filesets?

Reply:
This PR only supports Hive authorization. I created an issue #6133 to track it.